core-plugins/src/main/java/co/cask/hydrator/plugin/batch/action/SSHAction.java

```diff
@@ -116,49 +124,52 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
    */
   public static class SSHActionConfig extends PluginConfig {
     @Description("Command to be executed on the remote host. Should include filepath of script and any arguments")
+    @Macro
     private String command;
 
     @Description("Host name of the remote machine where the command needs to be executed.")
+    @Macro
     private String host;
 
     @Nullable
     @Description("Port to connect to. Defaults to 22")
+    @Macro
     private Integer port;
 
     @Description("User name used to connect to host")
+    @Macro
     private String user;
 
-    @Description("File path to Private key")
-    private String privateKeyFile;
+    @Description("Key used to lookup private key file from secure storage")
+    @Macro
+    private String privateKeyFileLookupKey;
 
     @Nullable
-    @Description("Password associated with private key")
-    private String password;
+    @Description("Passphrase associated with private key")
+    @Macro
+    private String passphrase;
 
     @Nullable
     @Description("Variable used to hold desired key name to be associated with script output in the instance " +
       "the script output contains information needed to be stored in the ActionContext for later use in the pipeline.")
+    @Macro
     private String output;
 
-    public SSHActionConfig() {
-      port = 22;
```

Comment:
@knazir so we use reflection to instantiate and set values. but internally it will call default constructor before trying to set values for the variables through reflection. so say the port was not provided in the config, due to this default constructor being called, this config will have a value of `22` for port and whatever you set inside.